rasa/nlu/classifiers/sklearn_intent_classifier.py

```diff
@@ -1,72 +1,92 @@
+
+# This is a workaround around until we have all components migrated to `GraphComponent`.
+SklearnIntentClassifier = SklearnIntentClassifier
+
 logger = logging.getLogger(__name__)
 
 if typing.TYPE_CHECKING:
     import sklearn
 
 
-class SklearnIntentClassifier(IntentClassifier):
-    """Intent classifier using the sklearn framework"""
+class SklearnIntentClassifierGraphComponent(GraphComponent):
+    """Intent classifier using the sklearn framework."""
 
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        return [DenseFeaturizer]
-
-    defaults = {
-        # C parameter of the svm - cross validation will select the best value
-        "C": [1, 2, 5, 10, 20, 100],
-        # gamma parameter of the svm
-        "gamma": [0.1],
-        # the kernels to use for the svm training - cross validation will
-        # decide which one of them performs best
-        "kernels": ["linear"],
-        # We try to find a good number of cross folds to use during
-        # intent training, this specifies the max number of folds
-        "max_cross_validation_folds": 5,
-        # Scoring function used for evaluating the hyper parameters
-        # This can be a name or a function (cfr GridSearchCV doc for more info)
-        "scoring_function": "f1_weighted",
-    }
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """The component's default config (see parent class for full docstring)."""
+        return {
+            # C parameter of the svm - cross validation will select the best value
+            "C": [1, 2, 5, 10, 20, 100],
+            # gamma parameter of the svm
+            "gamma": [0.1],
+            # the kernels to use for the svm training - cross validation will
+            # decide which one of them performs best
+            "kernels": ["linear"],
+            # We try to find a good number of cross folds to use during
+            # intent training, this specifies the max number of folds
+            "max_cross_validation_folds": 5,
+            # Scoring function used for evaluating the hyper parameters
+            # This can be a name or a function (cfr GridSearchCV doc for more info)
+            "scoring_function": "f1_weighted",
+        }
 
     def __init__(
         self,
-        component_config: Optional[Dict[Text, Any]] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         clf: "sklearn.model_selection.GridSearchCV" = None,
         le: Optional["sklearn.preprocessing.LabelEncoder"] = None,
     ) -> None:
         """Construct a new intent classifier using the sklearn framework."""
         from sklearn.preprocessing import LabelEncoder
 
-        super().__init__(component_config)
+        self.component_config = config
+        self._model_storage = model_storage
+        self._resource = resource
+        self._execution_context = execution_context
```

Comment:
do we actually need to save this one?

Reply:
True, `execution_context` is not required, I'll remove it; the others are needed.